pallets/detach/src/lib.rs

```diff
@@ -486,7 +543,7 @@ pub mod pallet {
 					// We can still have multiple transactions compete for the same "spot",
 					// and the one with higher priority will replace other one in the pool.
 					.and_provides((
-						data.hash.clone(),
+						data.merkle_root, // TODO Review - what if by co-incidence two collections that contain different things have the same merkle-root?
```

Comment:
With a 256 bit hash function, the probability of a merkle root collision is likely to be probabilistically impossible.

Reply:
To answer his comment.
Stars would collide and we would be sucked by a black hole if that happens.

Reply:
It's blake2 no?

Reply:
> It's blake2 no?

the merkle root is calculated using keccak256

Reply:
remove comment btw